src/template/js/sdk/firebolt.mjs

```diff
@@ -17,14 +17,50 @@
  */
 
 import { setMockResponses } from './Transport/MockTransport.mjs'
+import { registerAPI } from './Extensions/index.mjs'
 
 /* ${MOCK_IMPORTS} */
 
 setMockResponses({
   /* ${MOCK_OBJECTS} */
 })
 
+registerAPI('authorize', (grants) => {
```

Comment:
This is related to my question here:
https://github.com/rdkcentral/firebolt-openrpc/pull/52/files#r946241745

How does this authorize method work? Seems like it triggers the grants but then doesn't return anything back to the app that can be used as proof that the user said yes to the grant. For grants that have lifetimes I guess this would work because if they call token immediately after authorize then it will still be within the lifetime. But for grants with lifetimes as "single-use", this won't work.
I think I would expect authorize and token to be atomic for this reason. The grant happens and then they get a token as a result.

Reply:
Okay... the single use grant type definitely impacts this design.

I've been debating on having `authorize` just return a token. However, I think we want to support a use case where an extension SDK says:

```javascript
authorize(['a', 'b', 'c'])
```

1. The REST API only requires A, and B & C are optional.
2. The user only grants `a` and `c`.

So the Extension SDK would need to do something like:

```javascript
const grant = authorize(['a', 'b', 'c'])

if (grant.capabilities.includes('a')) {
  callRestApi('/foo/?token=' + grant.token) 
}
```

Reply:
Maybe we shouldn't even have single use tokens since they don't work well with the FAT. I guess we would just have small lifetime grants like 5 minutes